sycl/source/detail/device_global_map_entry.cpp

```diff
@@ -67,14 +66,21 @@ DeviceGlobalUSMMem &DeviceGlobalMapEntry::getOrAllocateDeviceGlobalUSM(
          "USM allocation for device and context already happened.");
   DeviceGlobalUSMMem &NewAlloc = NewAllocIt.first->second;
 
-  // Zero-initialize here and save the event.
+  // Initialize here and save the event.
   {
-    std::lock_guard<std::mutex> Lock(NewAlloc.MZeroInitEventMutex);
+    std::lock_guard<std::mutex> Lock(NewAlloc.MInitEventMutex);
     sycl::detail::pi::PiEvent InitEvent;
-    MemoryManager::fill_usm(NewAlloc.MPtr, QueueImpl, MDeviceGlobalTSize, 0,
-                            std::vector<sycl::detail::pi::PiEvent>{},
-                            &InitEvent);
-    NewAlloc.MZeroInitEvent = InitEvent;
+    // C++ guarantees members appear in memory in the order they are declared,
+    // so since the member variable that contains the initial contents of the
+    // device_global is right after the usm_ptr member variable we can do
+    // some pointer arithmetic to memcopy over this value to the usm_ptr. This
+    // value inside of the device_global will be zero-initialized if it was not
+    // given a value on construction.
+    MemoryManager::copy_usm(
+        (const uintptr_t *)MDeviceGlobalPtr + sizeof(MDeviceGlobalPtr),
```

Comment:
This ends up shifting the pointer by sizeof(uintptr_t *) * sizeof(MDeviceGlobalPtr) instead of just sizeof(MDeviceGlobalPtr). That's what causing the failures on Windows.
The pointer should be cast to just uintptr_t:
```suggestion
        reinterpret_cast<const void*>(reinterpret_cast<uintptr_t>(MDeviceGlobalPtr) + sizeof(MDeviceGlobalPtr)),
```